### ISSUE TITLE
pdm add: ModuleNotFoundError: No module named 'wheel.pep425tags'

### ISSUE THREAD

**Issue by pbryan** [frostming/pdm#135]
- [X] I have searched the issue tracker and believe that this is not a duplicate.

I'm seeing a different bug when running pdm on a standard Arch Linux installation.

## Actual behavior

```
pbryan@dynamo:~$ mkdir test
pbryan@dynamo:~$ cd test
pbryan@dynamo:~/test$ pdm init
Creating a pyproject.toml for PDM...
Please enter the Python interpreter to use: /usr/bin/python3.8
Using Python interpreter: /usr/bin/python3.8 (3.8.5)
Project name [test]: 
Project version [0.0.0]: 
License(SPDX name) [MIT]: 
Author name [Paul Bryan]: 
Author email [pbryan@anode.ca]: 
Python requires('*' to allow any) [>=3.8]: 
Changes are written to pyproject.toml.
pbryan@dynamo:~/test$ pdm add -v pytest
Adding packages to default dependencies: pytest
⠙ Resolving dependencies	Adding requirement pytest
Traceback (most recent call last):
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/resolver/core.py", line 21, in _merge_into_criterion
    crit = self.state.criteria[name]
KeyError: 'pytest'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pbryan/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 68, in __call__
    return self.main(*args, **kwargs)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 103, in main
    raise err.with_traceback(traceback)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 99, in main
    f(options.project, options)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/commands/add.py", line 35, in handle
    actions.do_add(
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/actions.py", line 140, in do_add
    resolved = do_lock(project, strategy, tracked_names, reqs)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/actions.py", line 58, in do_lock
    mapping, dependencies, summaries = resolve(
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/resolver/core.py", line 146, in resolve
    result = resolver.resolve(requirements)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/resolvelib/resolvers.py", line 427, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/resolvelib/resolvers.py", line 294, in resolve
    name, crit = self._merge_into_criterion(r, parent=None)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/resolver/core.py", line 23, in _merge_into_criterion
    crit = Criterion.from_requirement(self._p, requirement, parent)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/resolvelib/resolvers.py", line 83, in from_requirement
    candidates = list(candidates)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/resolver/providers.py", line 116, in find_matches
    yield from super().find_matches(requirements)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/resolver/providers.py", line 43, in find_matches
    candidates = self.repository.find_candidates(
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/models/repositories.py", line 206, in find_candidates
    with self.environment.get_finder(sources) as finder, allow_all_wheels():
  File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/models/environment.py", line 255, in get_finder
    finder = get_finder(
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/utils.py", line 241, in get_finder
    finder = get_package_finder(
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/utils.py", line 158, in get_package_finder
    abi = get_abi_tag(python_version)
  File "/home/pbryan/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/utils.py", line 59, in get_abi_tag
    from wheel.pep425tags import get_config_var, get_abbr_impl, get_flag
ModuleNotFoundError: No module named 'wheel.pep425tags'
```
## Environment Information

```
pbryan@dynamo:~/test$ pdm info && pdm info --env
PDM version:        0.8.6                     
Python Interpreter: /usr/bin/python3.8 (3.8.5)
Project Root:       /home/pbryan/test         
{
  "implementation_name": "cpython",
  "implementation_version": "3.8.5",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.8.1-arch1-1",
  "platform_system": "Linux",
  "platform_version": "#1 SMP PREEMPT Wed, 12 Aug 2020 18:50:43 +0000",
  "python_full_version": "3.8.5",
  "platform_python_implementaiton": "CPython",
  "python_version": "3.8",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
It is identified as dependency issue, try downgrading `wheel` to `<0.35` as a workaround

**Comment by pbryan** [comment]
I can confirm that downgrading to wheel-0.34.2 has pdm working correctly. Thanks.

**Comment by BrandonLWhite** [comment]
I'm running into this as well.

It looks like pdm is using a module internal to wheel:
https://github.com/frostming/pdm/blob/765f6025aab58e0d63fa241b64495961f42d68f6/pdm/utils.py#L59 

And of course it got removed in wheel 0.35 :(

**Comment by yevhenshymotiuk** [comment]
@frostming are you going to change https://github.com/frostming/pdm/blob/00336148c84f2fa435918240f647e21b0485e350/pyproject.toml#L99
to `requires = ["intreehooks", "wheel<0.35"]`?

**Comment by frostming** [comment]
@yevhenshymotiuk Not really, I would rather update the import statement to support both.

**Comment by frostming** [comment]
Fixed and released 0.8.7
